activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -418,8 +420,12 @@ def reconstruct_from_schema(db_config, format = ActiveRecord.schema_format, file
         load_schema(db_config, format, file)
       end
 
-      def dump_schema(db_config, format = ActiveRecord.schema_format) # :nodoc:
+      def dump_schema(db_config, format = nil) # :nodoc:
         require "active_record/schema_dumper"
+
+        ActiveSupport::Deprecation.warn("`format` will be removed as a parameter in future versions; it should be included in the passed `db_config`") if format
```

Comment:
We have 2 kinds of private methods in Rails, those under the private namespace and those that are undocumented, indicating that they are internal not to be consumed by applications. This method has `nodoc` which means it's internal and we don't need to handle deprecations.

```suggestion
```